tests/model_registry/model_catalog/upgrade/test_catalog_upgrade.py

```diff
@@ -0,0 +1,351 @@
+                "Default catalog",
+                id="test_verify_default_catalog_preserved_pre_upgrade",
+            ),
+        ],
+    )
+    @pytest.mark.pre_upgrade
+    def test_verify_catalog_exists_pre_upgrade(
+        self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        catalog_id: str,
+        catalog_description: str,
+    ):
+        """Verify catalog exists in sources"""
+        sources_response = get_sources_response_with_retry(
+            model_catalog_rest_url=model_catalog_rest_url, model_registry_rest_headers=model_registry_rest_headers
+        )
+
+        catalog_found = any(source["id"] == catalog_id for source in sources_response["items"])
+        assert catalog_found, (
+            f"{catalog_description} {catalog_id} not found in sources: {[s['id'] for s in sources_response['items']]}"
+        )
+
+    @pytest.mark.pre_upgrade
+    def test_verify_custom_model_accessible_pre_upgrade(
+        self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+    ):
+        """Verify each model in custom catalog is accessible"""
+        # Test first model from TEST_DATA (keeping it simple with one assert)
+        model_name = TEST_DATA["models"][0]
+        try:
+            verify_model_accessibility(
+                model_catalog_rest_url=model_catalog_rest_url,
+                model_registry_rest_headers=model_registry_rest_headers,
+                catalog_id=TEST_DATA["catalog_id"],
+                model_name=model_name,
+            )
+        except ResourceNotFoundError as e:
+            pytest.fail(f"Model {model_name} not found in custom catalog {TEST_DATA['catalog_id']}: {e}")
+
+    @pytest.mark.pre_upgrade
+    def test_verify_catalog_reconciliation_integrity_pre_upgrade(
+        self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+    ):
+        """Verify both catalogs are present in configmap with proper integrity"""
+        current_configmap_data = (
+            ConfigMap(name=DEFAULT_MODEL_CATALOG, client=admin_client, namespace=model_registry_namespace).instance.data
+            or {}
+        )
+
+        # Parse the current sources.yaml to verify structure
+        current_catalogs = yaml.safe_load(current_configmap_data[SOURCES_YAML_KEY])[CATALOGS_KEY]
+        catalog_ids = [catalog["id"] for catalog in current_catalogs]
+
+        # Single assertion: verify we have at least default and custom catalogs
+        assert len(catalog_ids) >= 2 and DEFAULT_CATALOG_ID in catalog_ids and TEST_DATA["catalog_id"] in catalog_ids, (
+            f"Should have at least default and custom catalogs. Found: {catalog_ids}"
+        )
+
+
+@pytest.mark.usefixtures(
+    "model_registry_namespace", "admin_client", "model_catalog_rest_url", "model_registry_rest_headers"
+)
+class TestPostUpgradeCatalog:
+    @pytest.mark.post_upgrade
```

Comment:
Where do we undo the custom catalog that we created in pre-upgrade?

Reply:
We must undo - in a fixture. Also, if you are not addressing a comment, let's not resolve it.

Reply:
It is already handled https://github.com/opendatahub-io/opendatahub-tests/pull/696/files#diff-9474382b27ff63e456408a1a0cef907e71a90a36d0eb36ea084537ae8e694981R211